.github/workflows/test-zkvm.yml

```diff
@@ -63,77 +63,68 @@ jobs:
+          else
+            CACHED_GIT_REV="$GIT_REV"
+          fi
+
           IMAGE_VERSION="$ZKVM_CRATE_VERSION-${GIT_REV:0:7}"
+          CACHED_IMAGE_VERSION="$ZKVM_CRATE_VERSION-${CACHED_GIT_REV:0:7}"
 
-          BASE_IMAGE_NAME="ghcr.io/${{ github.repository }}/ere-base"
-          BASE_ZKVM_IMAGE_NAME="ghcr.io/${{ github.repository }}/ere-base-${{ inputs.zkvm }}"
-          CLI_ZKVM_IMAGE_NAME="ghcr.io/${{ github.repository }}/ere-cli-${{ inputs.zkvm }}"
-
-          BASE_IMAGE_TAG="$BASE_IMAGE_NAME:$IMAGE_VERSION"
-          BASE_ZKVM_IMAGE_TAG="$BASE_ZKVM_IMAGE_NAME:$IMAGE_VERSION"
-          CLI_ZKVM_IMAGE_TAG="$CLI_ZKVM_IMAGE_NAME:$IMAGE_VERSION"
-
-          BUILD_BASE_IMAGE="true"
-          BUILD_BASE_ZKVM_IMAGE="true"
-
-          # Try to reuse image by re-tagging if related files are unchanged
-          if [ "${{ github.event_name }}" == "pull_request" ]; then
-            PR_BASE_GIT_REV="${{ github.event.pull_request.base.sha }}"
-            PR_BASE_IMAGE_VERSION="$ZKVM_CRATE_VERSION-${PR_BASE_GIT_REV:0:7}"
-
-            if [ "${{ steps.changed_files.outputs.base_any_changed }}" == "false" ]; then
-              if docker buildx imagetools create --tag "$BASE_IMAGE_TAG" "$BASE_IMAGE_NAME:$PR_BASE_IMAGE_VERSION" 2>/dev/null; then
-                echo "Created new tag $BASE_IMAGE_TAG referring to $BASE_IMAGE_NAME:$PR_BASE_IMAGE_VERSION"
-                BUILD_BASE_IMAGE="false"
-              fi
-            fi
-
-            if [ "${{ steps.changed_files.outputs.base_zkvm_any_changed }}" == "false" ]; then
-              if docker buildx imagetools create --tag "$BASE_ZKVM_IMAGE_TAG" "$BASE_ZKVM_IMAGE_NAME:$PR_BASE_IMAGE_VERSION" 2>/dev/null; then
-                echo "Created new tag $BASE_ZKVM_IMAGE_TAG referring to $BASE_ZKVM_IMAGE_NAME:$PR_BASE_IMAGE_VERSION"
-                BUILD_BASE_ZKVM_IMAGE="false"
-              fi
-            fi
-          fi
+          IMAGE_REPO="ghcr.io/${{ github.repository }}"
+          BASE_IMAGE_TAG="$IMAGE_REPO/ere-base:$CACHED_IMAGE_VERSION"
+          BASE_ZKVM_IMAGE_TAG="$IMAGE_REPO/ere-base-${{ inputs.zkvm }}:$CACHED_IMAGE_VERSION"
+          CLI_ZKVM_IMAGE_TAG="$IMAGE_REPO/ere-cli-${{ inputs.zkvm }}:$CACHED_IMAGE_VERSION"
 
           echo "image_version=$IMAGE_VERSION" >> $GITHUB_OUTPUT
           echo "base_image_tag=$BASE_IMAGE_TAG" >> $GITHUB_OUTPUT
           echo "base_zkvm_image_tag=$BASE_ZKVM_IMAGE_TAG" >> $GITHUB_OUTPUT
           echo "cli_zkvm_image_tag=$CLI_ZKVM_IMAGE_TAG" >> $GITHUB_OUTPUT
-          echo "build_base_image=$BUILD_BASE_IMAGE" >> $GITHUB_OUTPUT
-          echo "build_base_zkvm_image=$BUILD_BASE_ZKVM_IMAGE" >> $GITHUB_OUTPUT
 
       - name: Build ere-base image
-        if: ${{ steps.inspect_image.outputs.build_base_image == 'true' }}
+        if: (github.event_name == 'push' && github.ref == 'refs/heads/master') || (github.event.pull_request.head.repo.full_name == github.repository && steps.changed_files.outputs.any_changed == 'true')
```

Comment:
Build if it is push to `master`, or if the PR is from `eth-act/ere` (which has the permission to push to ghcr) and the base images related files are changed (the latter helps to reduce the CI time a bit).